core/src/context.rs

```diff
@@ -32,6 +39,68 @@ use std::{
     time::Duration,
 };
 
+/// This is a local mock for the `agent/sign` conductor API function.
+/// It creates a syntactically equivalent signature using dpki::Keypair
+/// but with key generated from a static/deterministic mock seed.
+/// This enables unit testing of core code that creates signatures without
+/// depending on the conductor or actual key files.
+pub fn mock_signer(payload: String) -> String {
+    // Create deterministic seed:
+    let mut seed = SecBuf::with_insecure(SEEDSIZE);
+    let mut mock_seed: Vec<u8> = Vec::new();
+    for i in 1..SEEDSIZE {
+        mock_seed.push(i as u8);
+    }
+    seed.write(0, mock_seed.as_slice())
+        .expect("SecBuf must be writeable");
+
+    // Create keypair from seed:
+    let mut keypair = Keypair::new_from_seed(&mut seed).unwrap();
+
+    // Convert payload string into a SecBuf
+    let payload = payload.as_bytes();
+    let mut message = SecBuf::with_insecure(payload.len());
+    message.write(0, payload).expect("SecBuf must be writeable");
+
+    // Create signature
+    let mut message_signed = SecBuf::with_insecure(64);
```

Comment:
is there a constant on the crate for signature size so we don't have to use magic number 64?

Reply:
Apparently there is not.